models/commons/src/reference/types.ts

```diff
@@ -4,12 +4,12 @@ export type TReferenceBuilder<TypeId = string> = TBuilder<TReference<TypeId>>;
 
 export interface TReference<TypeId = string> {
   typeId: TypeId;
-  id: string;
-  key: string;
+  id?: string;
+  key?: string;
 }
 
 export type TExpandedReferenceObject<TypeId = string> = {
-  id: TReference<TypeId>['id'];
+  id?: TReference<TypeId>['id'];
```

Comment:
cc @emmenko @stephsprinkle 

This is the big change here. I made `id` and `key` optional fields, as we need a way to create an `empty` preset for reference types. The generator for this spoofs an `id`, which means if we send these to the CoCo API, we could get id/key collisions or errors. I saw two paths forward:

1. have the generator no longer spoof an `id`. This means we need to alter every implementation of `reference` to include a spoofed id. A lot of work, to be honest
2. Make both fields optional, and include an `empty` preset so we can clear existing `id`'s and add our own `key`. This is what I chose to do. 

This solution passed typecheck and all tests, so I felt it stable.

Reply:
They other option is to create a KeyReference model and leave this one be just `id`.

Reply:
Actually nvm, rather deal with it here. Don't want the edits to `Reference` merged in

Reply:
Can the optional stuff be removed now that we have `KeyReference`?

Reply:
Thanks for the reminder, goldfish memory over here

8a5b631dd1465de561e040ec9d27c1128f9c79ce